pandas/core/groupby.py

```diff
@@ -296,6 +297,57 @@ def groups(self):
         return self.grouper.groups
 
 
+class OrderedGrouper(Grouper):
+
+    def __init__(self, proportions=(1,1), axis=None):
+        self._proportions = proportions
+        self._axis = axis
+        self.key = None
+        # check weight type
+        if len(self._proportions) < 2:
+            raise ValueError("must split into more than 1 partition")
+        for w in self._proportions:
+            if not isinstance(w, Real) or w <=0:
+                raise ValueError("weights must be strictly positive real numbers")
+
+        weights_total = reduce(lambda x, y: x+y, self._proportions, 0)
+
```

Comment:
e.g. ``self._proportions.sum()`` (look at ``DataFrame.sample`` for some example of handling these types of weights)